src/azul/auth.py

```diff
@@ -1,7 +1,19 @@
 import abc
+from inspect import (
+    isabstract,
+)
+
 import attr
 
+from azul import (
+    JSON,
+)
+from azul.json import (
+    copy_json,
+)
+
 
+@attr.s(auto_attribs=True, frozen=True)
```

Comment:
`kwonly=True`, correct?

Reply:
Decided at PL to remove the decorator entirely.

Reply:
Realized we actually do benefit from the decorator since it avoids type warnings. Leaving as-is for now.

Reply:
Note that `kw_only=True` is omitted from the subclasses